sound/soc/sof/ipc4-topology.c

```diff
@@ -1485,6 +1485,59 @@ static int sof_ipc4_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget
 	return ret;
 }
 
+static int sof_ipc4_get_queue_id(struct snd_sof_widget *swidget, int bind_type,
+				 const char *peer_name)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct snd_sof_pin_binding *pin_binding;
+	struct ida *queue_ida;
+
+	int num_pins;
+	int i;
+
+	if (swidget->num_pin_bindings > 0) {
+		for (i = 0; i < swidget->num_pin_bindings; i++) {
+			pin_binding = &swidget->pin_binding[i];
+
+			if (pin_binding->bind_type == bind_type &&
+			    !strcmp(pin_binding->wname, peer_name)) {
+				return pin_binding->queue_id;
+			}
+		}
+		dev_err(scomp->dev, "failed to find queue id from pin binding array for %s\n",
+			swidget->widget->name);
+		return -EINVAL;
+	}
+
```

Comment:
so the code below is used if there is no binding specified in topology?

please add comments!